homeassistant/components/screenlogic/climate.py

```diff
@@ -0,0 +1,217 @@
+        super().__init__(coordinator, body)
+        self._configured_heat_modes = []
+        # Is solar listed as available equipment?
+        if self.coordinator.data["config"]["equipment_flags"] & EQUIPMENT.FLAG_SOLAR:
+            self._configured_heat_modes.extend(
+                [HEAT_MODE.SOLAR, HEAT_MODE.SOLAR_PREFERRED]
+            )
+        self._configured_heat_modes.append(HEAT_MODE.HEATER)
+        self._last_preset = None
+
+    @property
+    def name(self) -> str:
+        """Name of the heater."""
+        ent_name = self.body["heat_status"]["name"]
+        return f"{self.gateway_name} {ent_name}"
+
+    @property
+    def state(self) -> str:
+        """State of the heater."""
+        return self.hvac_mode
+
+    @property
+    def min_temp(self) -> float:
+        """Minimum allowed temperature."""
+        return self.body["min_set_point"]["value"]
+
+    @property
+    def max_temp(self) -> float:
+        """Maximum allowed temperature."""
+        return self.body["max_set_point"]["value"]
+
+    @property
+    def current_temperature(self) -> float:
+        """Return water temperature."""
+        return self.body["last_temperature"]["value"]
+
+    @property
+    def target_temperature(self) -> float:
+        """Target temperature."""
+        return self.body["heat_set_point"]["value"]
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement."""
+        if self.config_data["is_celcius"]["value"] == 1:
+            return TEMP_CELSIUS
+        return TEMP_FAHRENHEIT
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return the current hvac mode."""
+        if self.body["heat_mode"]["value"] > 0:
+            return HVAC_MODE_HEAT
+        return HVAC_MODE_OFF
+
+    @property
+    def hvac_modes(self):
+        """Return th supported hvac modes."""
+        return SUPPORTED_MODES
+
+    @property
+    def hvac_action(self) -> str:
+        """Return the current action of the heater."""
+        if self.body["heat_status"]["value"] > 0:
+            return CURRENT_HVAC_HEAT
+        if self.hvac_mode == HVAC_MODE_HEAT:
+            return CURRENT_HVAC_IDLE
+        return CURRENT_HVAC_OFF
+
+    @property
+    def preset_mode(self) -> str:
+        """Return current/last preset mode."""
+        if self.hvac_mode == HVAC_MODE_OFF:
+            return HEAT_MODE.NAME_FOR_NUM[self._last_preset]
+        return HEAT_MODE.NAME_FOR_NUM[self.body["heat_mode"]["value"]]
+
+    @property
+    def preset_modes(self):
+        """All available presets."""
+        return [
+            HEAT_MODE.NAME_FOR_NUM[mode_num] for mode_num in self._configured_heat_modes
+        ]
+
+    @property
+    def supported_features(self):
+        """Supported features of the heater."""
+        return SUPPORTED_FEATURES
+
+    async def async_set_temperature(self, **kwargs) -> None:
+        """Change the setpoint of the heater."""
+        temperature = kwargs[ATTR_TEMPERATURE]
```

Comment:
```suggestion
        temperature = kwargs.get(ATTR_TEMPERATURE)
```

Probably raise if `temperature` is `None` since nothing else can happen here

Reply:
`raise` or `return`? I see quite a few integrations just `return` if `temperature` is `None`.

Reply:
If it's not going to do what the user expects we should raise since they will not get any feedback otherwise.

Reply:
Commit https://github.com/home-assistant/core/pull/48136/commits/f077730806061630259f1bb29d7e6049203f806d